src/core/thread/key_manager.hpp

```diff
@@ -85,7 +85,7 @@ class SecurityPolicy : public otSecurityPolicy, public Equatable<SecurityPolicy>
      * and Security Policy Flags.
      *
      */
-    SecurityPolicy(void) { SetToDefault(); }
+    SecurityPolicy(void) : otSecurityPolicy() { SetToDefault(); }
```

Comment:
This may not do what we intend in all cases (all toochains) since `otSecuretyPolicy()` is a C struct definition (so no constructor is defined for it). Default constructor may not clear all its fields (behavior can be toolchain dependent)

It may be safer to explicitly clear all its bytes. Here is my suggestion:
- We can add `public Clearable<SecurityPolicy>` as a base class to `class SecurityPolicy`:
```c++
class SecurityPolicy : public otSecurityPolicy, public Equatable<SecurityPolicy>, public Clearable<SecurityPolicy>

```
- And then we can add a `Clear()` call in `SetToDefault()` method

Reply:
@abtink applied. let me know if that looks right